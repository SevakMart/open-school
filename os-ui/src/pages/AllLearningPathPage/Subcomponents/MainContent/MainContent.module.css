.mainContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 100%;
}
.mainContainer .learningPathContainer {
  position: absolute;
  top: 100px;
  left: 400px;
  width: 65%;
  overflow: hidden;
}/*# sourceMappingURL=MainContent.module.css.map */