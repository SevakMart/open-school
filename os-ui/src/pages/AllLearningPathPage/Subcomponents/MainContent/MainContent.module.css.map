{"version":3,"sources":["MainContent.module.scss","../../../../styles/Mixins/FlexMixins.scss","MainContent.module.css"],"names":[],"mappings":"AAEA;ECDE,aAAA;EACA,mBDCqB;ECArB,mBAAA;EACA,8BDDkC;EAChC,WAAA;AEEJ;AFAI;EACI,kBAAA;EACA,UAAA;EACA,WAAA;EACA,UAAA;EACA,gBAAA;AEER","file":"MainContent.module.css"}