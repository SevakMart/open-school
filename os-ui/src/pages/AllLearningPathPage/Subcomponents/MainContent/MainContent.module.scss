@import "../../../../styles/Mixins/FlexMixins.scss";


.mainContainer{
    @include flexStyle(row,center,space-between);
    .learningPathContainer {
    width:75%;
    align-self: flex-start;
    margin:56px 88px 56px 54px;
   }
}