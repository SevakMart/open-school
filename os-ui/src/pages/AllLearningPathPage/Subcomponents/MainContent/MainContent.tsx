import FilterComponent from '../FilterComponent/FilterComponent';
import CourseContentHeader from '../CourseCountentHeader/CourseContentHeader';
import styles from './MainContent.module.scss';

const MainContent = () => {
  // const {} = styles;
  const a = 1;
  return (
    <div>
      <FilterComponent />
      <div>
        <CourseContentHeader />
      </div>
    </div>
  );
};
export default MainContent;
