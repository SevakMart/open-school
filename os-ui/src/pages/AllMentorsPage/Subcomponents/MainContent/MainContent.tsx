import { useState } from 'react';
import AllMentorsPageHeader, { Mentors } from '../AllMentorsPageHeader/AllMentorsPageHeader';
import Content from '../Content/Content';
import SavedMentors from '../SavedMentors/SavedMentors';
import styles from './MainContent.module.scss';

/* eslint-disable max-len */

const MainContent = () => {
  const [focusedHeader, setFocusedHeader] = useState(Mentors.ALL_MENTORS);
  const { mainContentContainer } = styles;
  return (
    <div className={mainContentContainer}>
      <AllMentorsPageHeader changeHeaderFocus={(headerTitle) => setFocusedHeader(headerTitle)} />
      {focusedHeader === Mentors.ALL_MENTORS ? <Content /> : <SavedMentors />}
    </div>
  );
};
export default MainContent;
