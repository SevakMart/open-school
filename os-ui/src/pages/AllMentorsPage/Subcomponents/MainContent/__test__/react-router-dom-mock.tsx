export const mockUseNavigate = jest.fn();
export const reactRouterDomMock = {
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUseNavigate,
  useLocation: () => ({ search: '' }),
};
