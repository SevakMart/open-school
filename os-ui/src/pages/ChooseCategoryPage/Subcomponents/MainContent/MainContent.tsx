import CategoryWithSubcategoriesProfile from '../CategoryWithSubcategoriesProfile/CategoryWithSubcategoriesProfile';
import Loader from '../../../../component/Loader/Loader';
import { SearchedCategoryType } from '../../../../types/SearchedCategoryType';
import { ErrorField } from '../../../../component/ErrorField/ErrorField';
import styles from './MainContent.module.scss';

/* eslint-disable max-len */

const MainContent = ({ isLoading, errorMessage, searchedCategories }:{isLoading:boolean, errorMessage:string, searchedCategories:SearchedCategoryType}) => {
  const { categoriesList } = styles;
  return (
    <>
      <div className={categoriesList}>
        {isLoading && <Loader />}
        {errorMessage !== '' && <ErrorField.MainErrorField className={['subcategoriesErrorField']}>{errorMessage}</ErrorField.MainErrorField>}
        {errorMessage === '' && (
          Object.entries(searchedCategories).map((category) => (
            <CategoryWithSubcategoriesProfile
              key={category[0]}
              parentCategory={category[0]}
              subcategories={category[1]}
            />
          ))
        )}
      </div>
    </>
  );
};
export default MainContent;
