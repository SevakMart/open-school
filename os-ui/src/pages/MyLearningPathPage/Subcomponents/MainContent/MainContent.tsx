import { useSelector } from 'react-redux';
import { RootState } from '../../../../redux/Store';
import { UserCourseType } from '../../../../types/CourseTypes';
import InProgressCourse from '../InProgressCourse/InProgressCourse';
import CompletedCourse from '../CompletedCourse/CompledtedCourse';
import ContentRenderer from '../../../../component/ContentRenderer/ContentRenderer';
import styles from './MainContent.module.scss';

/* eslint-disable max-len */

const MyLearningPathMainContent = () => {
  const userEnrolledCourseByCourseStatus = useSelector<RootState>((state) => state.userEnrolledCourseByCourseStatus);
  const { entity, isLoading, errorMessage } = userEnrolledCourseByCourseStatus as {entity:any[], isLoading:boolean, errorMessage:string};
  const { courseContainer } = styles;
  return (
    <div className={courseContainer}>
      <ContentRenderer
        isLoading={isLoading}
        errorMessage={errorMessage}
        entity={entity}
        errorFieldClassName="myLearningPathsErrorStyle"
        isMyLearningPathPage
        render={(entity) => (
          entity.map((userCourse:UserCourseType, index:number) => {
            if (userCourse.courseStatus === 'IN_PROGRESS') {
              return (
                <InProgressCourse
                  key={index}
                  title={userCourse.title}
                  courseStatus={userCourse.courseStatus}
                  percentage={userCourse.percentage}
                  remainingTime={userCourse.remainingTime}
                  dueDate={userCourse.dueDate}
                  courseId={userCourse.courseId}
                  id={userCourse.courseId}
                />
              );
            }
            return (
              <CompletedCourse
                key={index}
                title={userCourse.title}
                courseStatus={userCourse.courseStatus}
                grade={userCourse.grade}
              />
            );
          })
        )}
      />
    </div>
  );
};
export default MyLearningPathMainContent;
