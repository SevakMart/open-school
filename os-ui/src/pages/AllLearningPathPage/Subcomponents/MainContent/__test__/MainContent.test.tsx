import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { store as myStore } from '../../../../../redux/Store';
import { reactRouterDomMock } from '../../../../AllMentorsPage/Subcomponents/MainContent/__test__/react-router-dom-mock';
import MainContent from '../MainContent';

jest.doMock('react-router-dom', () => reactRouterDomMock);

describe('Create test cases for CourseContentHeader', () => {
  test('Make a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={myStore}>
        <MemoryRouter>
          <MainContent />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
