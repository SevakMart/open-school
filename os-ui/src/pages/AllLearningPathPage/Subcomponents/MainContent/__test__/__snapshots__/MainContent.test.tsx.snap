// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for CourseContentHeader Make a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
    style="display: none;"
  >
    <div
      class="learningPathContainer"
    >
      <div
        class="mainCoreContainer"
      />
    </div>
  </div>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
